spec/requests/applicants_spec.rb

```diff
@@ -4,46 +4,132 @@
   describe 'POST applicants' do
     let(:assessment) { create :assessment }
     let(:applicant) { 'applicant' }
+    let(:headers) { { 'CONTENT_TYPE' => 'application/json' } }
     let(:params) do
       {
-        dummy_payload: true,
-        assessment_id: '123'
-      }.to_json
+        assessment_id: assessment.id,
+        applicant: {
+          date_of_birth: 20.years.ago.to_date,
+          involvement_type: 'Applicant',
+          has_parter_opponent: false,
+          receives_qualifying_benefit: true
+        }
+      }
     end
 
     context 'valid payload' do
       before do
-        service = double ApplicantCreationService, success?: true, applicant: applicant
-        expect(ApplicantCreationService).to receive(:call).with(params).and_return(service)
-        post assessment_applicant_path(assessment.id), params: params
+        # service = double ApplicantCreationService, success?: true, applicant: applicant
+        expect(ApplicantCreationService).to receive(:call).with(params.to_json).and_return(service)
+        post assessment_applicant_path(assessment.id), params: params.to_json, headers: headers
       end
 
-      it 'returns success' do
-        expect(response).to have_http_status(:success)
+      context 'service returns success' do
+        let(:service) { double ApplicantCreationService, success?: true, applicant: applicant }
+
+        it 'returns success', :show_in_doc do
+          expect(response).to have_http_status(:success)
+        end
+
+        it 'returns expected response' do
+          expect(json[:success]).to eq(true)
+          expect(json[:errors]).to be_empty
+          expect(json[:objects]).to eq(applicant)
+        end
       end
 
-      it 'returns expected response' do
-        expect(json[:success]).to eq(true)
-        expect(json[:errors]).to be_empty
-        expect(json[:objects]).to eq(applicant)
+      context 'service returns failure' do
+        let(:service) { double ApplicantCreationService, success?: false, errors: %w[error-1 error-2] }
+
+        it 'returns 422' do
+          expect(response).to have_http_status(:unprocessable_entity)
+        end
+
+        it 'returns expected response' do
+          expect(json[:success]).to eq(false)
+          expect(json[:errors]).to eq %w[error-1 error-2]
+          expect(json[:objects]).to be_nil
+        end
       end
     end
 
     context 'invalid payload' do
-      before do
-        service = double ApplicantCreationService, success?: false, errors: ['xxx']
-        expect(ApplicantCreationService).to receive(:call).with(params).and_return(service)
-        post assessment_applicant_path(assessment.id), params: params
+      shared_examples 'it fails with message' do |message|
```

Comment:
Good stuff. I didn't realise you could pas a parameter to a `shared_examples` block